Source/Python/WarpXWrappers.cpp

```diff
@@ -624,6 +624,7 @@ namespace
         {
             const long np = pti.numParticles();
             auto& wp = pti.GetAttribs(PIdx::w);
+            // Do this unconditionally from myspc.do_not_deposit, to support diagnostic uses
```

Comment:
A suggested change to the comment:

Do this unconditionally, ignoring myspc.do_not_deposit, to support diagnostic use